server/api/controller/plan.js

```diff
@@ -48,20 +46,29 @@ class PlanController extends Controller {
 			if(!GJV.valid(geojson)){
 				throw new Exception.BadRequest('point is invalid'); 
 			}
+
 			const polygon = wkt.convert(geojson);
 
 			// some databases (mysql 8) already support returning ST_Distance in meters
 			// if the geometries are in the same system of reference, and others
 			// (mysql 5.7, mariadb) return the distance in degrees and need to be
 			// multiplied to get an approximate meters value
-			if (Config.database.dbDistanceInMeters) {
-				q.columns.push(Knex.raw(`ST_Distance(geom, ST_GeomFromText("${polygon}",4326)) as distance`))
+			if (Config.locationSearch.dbDistanceInMeters) {
+				q.columns.push(Knex.raw(`ST_Distance(geom, ST_GeomFromText("${polygon}",4326)) as distance`));
 			} else {
-				q.columns.push(Knex.raw(`ST_Distance(geom, ST_GeomFromText("${polygon}",4326))*111195 as distance`))
+				q.columns.push(Knex.raw(`ST_Distance(geom, ST_GeomFromText("${polygon}",4326))*111195 as distance`));
 			}
 
-			q.orderByRaw = ['distance']
-			delete q.order
+			q.orderByRaw = ['distance'];
+			delete q.order;
+
+			if (Config.locationSearch.filterPlansRadiusKm !== null) {
+				if (Config.locationSearch.dbDistanceInMeters) {
+					q.whereRaw = [Knex.raw(`ST_Within(geom_centroid, ST_Buffer(ST_GeomFromText("${polygon}", 4326), ${Config.locationSearch.filterPlansRadiusKm}*1000))`)];
+				} else {
+					q.whereRaw = [Knex.raw(`ST_Within(geom_centroid, ST_Buffer(ST_GeomFromText("${polygon}", 4326), ${Config.locationSearch.filterPlansRadiusKm}*1000/111195))`)];
```

Comment:
its would be better if the division const is calculated beforehand, then injected to the raw where clause .
For instance, 
`
const divFactor= Config.locationSearch.dbDistanceInMeters? 1 : 111195;
q.whereRaw = [Knex.raw(`ST_Within(geom_centroid, ST_Buffer(ST_GeomFromText("${polygon}", 4326), ${Config.locationSearch.filterPlansRadiusKm}*1000/ ${divFactor}))`)];
`